test/integration/model/schema.test.js

```diff
@@ -0,0 +1,458 @@
+          var self = this;
+
+          var restaurauntModel = self.RestaurantOne.build({bar: 'one.1'});
+
+          return restaurauntModel.save()
+            .then(function() {
+              restaurauntModel = self.RestaurantOne.build({bar: 'one.2'});
+              return restaurauntModel.save();
+            }).then(function() {
+              restaurauntModel = self.RestaurantTwo.build({bar: 'two.1'});
+              return restaurauntModel.save();
+            }).then(function() {
+              restaurauntModel = self.RestaurantTwo.build({bar: 'two.2'});
+              return restaurauntModel.save();
+            }).then(function() {
+              restaurauntModel = self.RestaurantTwo.build({bar: 'two.3'});
+              return restaurauntModel.save();
+            }).then(function() {
+              return self.RestaurantOne.findAll();
+            }).then(function(restaurantsOne) {
+              expect(restaurantsOne).to.not.be.null;
+              expect(restaurantsOne.length).to.equal(2);
+              restaurantsOne.forEach(function(restaurant) {
+                expect(restaurant.bar).to.contain('one');
+              });
+              return self.RestaurantOne.findAndCountAll();
+            }).then(function(restaurantsOne) {
+              expect(restaurantsOne).to.not.be.null;
+              expect(restaurantsOne.rows.length).to.equal(2);
+              expect(restaurantsOne.count).to.equal(2);
+              restaurantsOne.rows.forEach(function(restaurant) {
+                expect(restaurant.bar).to.contain('one');
+              });
+              return self.RestaurantTwo.findAll();
+            }).then(function(restaurantsTwo) {
+              expect(restaurantsTwo).to.not.be.null;
+              expect(restaurantsTwo.length).to.equal(3);
+              restaurantsTwo.forEach(function(restaurant) {
+                expect(restaurant.bar).to.contain('two');
+              });
+              return self.RestaurantTwo.findAndCountAll();
+            }).then(function(restaurantsTwo) {
+              expect(restaurantsTwo).to.not.be.null;
+              expect(restaurantsTwo.rows.length).to.equal(3);
+              expect(restaurantsTwo.count).to.equal(3);
+              restaurantsTwo.rows.forEach(function(restaurant) {
+                expect(restaurant.bar).to.contain('two');
+              });
+            });
+        });
+      });
+
+      describe('Add data via instance.save, retrieve via model.count and model.find', function() {
+        it('should be able to insert data into both schemas using instance.save count it and retrieve it via findAll with where', function() {
+          var self = this;
+
+          var restaurauntModel = self.RestaurantOne.build({bar: 'one.1'});
+
+          return restaurauntModel.save().then(function() {
+            restaurauntModel = self.RestaurantTwo.build({bar: 'two.1'});
```

Comment:
I'm purposefully doing this in multiple steps to verify that built models maintain there proper $schema setting across calls as this was not the case with the original implementation because it was changing the schema option on the shared prototype.